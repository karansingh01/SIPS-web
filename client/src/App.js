"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
require("./App.css");
const client_1 = require("./api/client");
const randomDrink_1 = require("./api/graphql/randomDrink");
const alcoholFilter_1 = require("./api/graphql/alcoholFilter"); //klager p√• i konsollen
const react_router_dom_1 = require("react-router-dom");
const react_1 = require("@chakra-ui/react");
const AllCocktailsPage_1 = __importDefault(require("./pages/AllCocktailsPage"));
const CocktailRecipePage_1 = __importDefault(require("./pages/CocktailRecipePage"));
const FavoriteCocktailsPage_1 = __importDefault(require("./pages/FavoriteCocktailsPage"));
// format: response.data.<queryName>.drinks[0].<whateverYouWantToCollect>
// drinks[0] gives you the first drink in the array, which is the only drink since we
// are only querying for one random drink.
client_1.client.query({ query: randomDrink_1.RandomDrinkQuery }).then((response) => {
    console.log(response.data.randomDrink.drinks[0].strDrink);
});
client_1.client.query({ query: alcoholFilter_1.AlcoholFilterQuery }).then((response) => {
    console.log(response.data.alcoholFilter.drinks[0].strDrink);
});
(0, alcoholFilter_1.DrinkByAlc)("Vodka");
function App() {
    return (<react_1.ChakraProvider>
      <react_router_dom_1.HashRouter>
        <react_router_dom_1.Routes>
          <react_router_dom_1.Route path="/" element={<AllCocktailsPage_1.default />}/>
          <react_router_dom_1.Route path="/cr" element={<CocktailRecipePage_1.default />}/>
          <react_router_dom_1.Route path="/fav" element={<FavoriteCocktailsPage_1.default />}/>
        </react_router_dom_1.Routes>
      </react_router_dom_1.HashRouter>
    </react_1.ChakraProvider>
    /*       <DisplayLocations />
     */
    );
}
exports.default = App;
