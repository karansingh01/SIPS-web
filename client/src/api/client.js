"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const client_1 = require("@apollo/client");
const apollo_link_rest_1 = require("apollo-link-rest");
// Set rest-link endpoint
const restLink = new apollo_link_rest_1.RestLink({ uri: 'https://www.thecocktaildb.com/api/json/v1/1/' });
// Create apollo client
exports.client = new client_1.ApolloClient({
    cache: new client_1.InMemoryCache(),
    link: restLink,
});
