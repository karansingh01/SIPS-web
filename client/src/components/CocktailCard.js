"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@chakra-ui/react");
const react_2 = require("react");
const fa_1 = require("react-icons/fa");
function CocktailCard({ cocktail, }) {
    const IMAGE = cocktail.image;
    let name = cocktail.name;
    let description = cocktail.description;
    let favorite = cocktail.favorite;
    /**
     * toggler favoritt-knapp (ikke faktisk logikk sÃ¥ langt obv)
     */
    const [isFav, setFav] = (0, react_2.useState)(favorite);
    const toggleFav = () => {
        setFav(!isFav);
        cocktail.favorite = isFav;
    };
    return (<react_1.Center py={12}>
      <react_1.Box role={"group"} p={6} maxW={"330px"} w={"full"} rounded={15} bg={(0, react_1.useColorModeValue)("#f7f0e9", "gray.800")} borderColor={"beige"} pos={"relative"}>
        <react_1.Stack style={{ flexDirection: "row-reverse" }} marginBottom={-20}>
          {isFav ? (<fa_1.FaHeart color={"lightpink"} size={20} onClick={toggleFav}/>) : (<fa_1.FaRegHeart color={"lightpink"} size={20} onClick={toggleFav}/>)}
        </react_1.Stack>
        <react_1.Stack pt={20} align={"center"} marginBottom={10} marginTop={0}>
          <react_1.Text color={"gray.500"} fontSize={"lg"} textTransform={"uppercase"}>
            {name}
          </react_1.Text>
        </react_1.Stack>
        <react_1.Box alignContent={"center"}>
          <react_1.Image rounded={15} height={230} width={282} objectFit={"cover"} src={IMAGE}/>
        </react_1.Box>
        <react_1.Stack pt={10}>
          <react_1.Container maxW="270px" textAlign={"center"} fontSize={"15px"} fontFamily={"body"} fontWeight={100}>
            {description.slice(0, 20)}
            {description[21] ? "..." : null}
          </react_1.Container>
          <react_1.Text textTransform={"uppercase"} fontSize={"10px"} color={"gray.300"} textAlign={"right"} marginBottom={"10px"}>
            How I'm made...
          </react_1.Text>
        </react_1.Stack>
      </react_1.Box>
    </react_1.Center>);
}
exports.default = CocktailCard;
