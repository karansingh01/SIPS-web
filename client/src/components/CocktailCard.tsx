import {
    Box,
    Center,
    useColorModeValue,
    Heading,
    Text,
    Textarea,
    Stack,
    Image,
    HStack,
    IconButton,
    Container
  } from '@chakra-ui/react';
  import {FaRegHeart, FaHeart} from "react-icons/fa";
  
  /* const IMAGE =
    'https://www.liquor.com/thmb/fO-COKLw_iEA28v8K4XQjzMhkfw=/735x0/very-sexy-martini-720x720-primary-b1212ebf73f54f898a56f7f0b60c0a34.jpg';
   */
  export default function CocktailCard({name, description, favorite, image}:{name: string; description: string; favorite: boolean, image: string}) {
    const IMAGE = image;

    return (
      <Center py={12}>
        <Box
          role={'group'}
          p={6}
          maxW={'330px'}
          w={'full'}
          rounded={15}
          bg={useColorModeValue('lightyellow', 'gray.800')}
          pos={'relative'}>
            <Stack style={{flexDirection: 'row-reverse'}}> 
            {favorite ?  <FaHeart color={'lightpink'} size={20} /> : <FaRegHeart color={'lightpink'} size={20} />}
            </Stack>
            <Stack pt={20} align={'center'} marginBottom={10} marginTop={0}> 
                <Text color={'gray.500'} fontSize={'15px'}/*FontSize med stÃ¸rrekser (sm, lg, xl, ..) funker ikke */ textTransform={'uppercase'} >
                {name}
                </Text> 
            </Stack>
          <Box
          alignContent={'center'}
            /* marginTop={'20px'}
            mt={-12}
            pos={'relative'}
            color={'red'}
            borderRadius={'full'}
            height={'230px'}
            _after={{
              transition: 'all .3s ease',
              content: '""',
              w: 'full',
              h: 'full',
              pos: 'absolute',
              top: 5,
              left: 0,
              backgroundImage: `url(${IMAGE})`, 
              filter: 'blur(15px)',
              zIndex: -1,
            }}
            _groupHover={{
              _after: {
                filter: 'blur(20px)',
              },
            }} */>
            <Image
              rounded={15}
              height={230}
              width={282}
              objectFit={'cover'}
              src={IMAGE}
            />
          </Box>
          <Stack pt={10} >
            <Container maxW='270px' textAlign={'center'} fontSize={'15px'} fontFamily={'body'} fontWeight={100}>
              {description} {/* blablablablabla blabalablabla blablabla eruygierguouerwg ergag aegrgeg aergeargregegrag */}
            </Container>
              <Text textTransform={'uppercase'} fontSize={'10px'} color={'gray.300'} textAlign={'right'} marginBottom={'10px'}>
                How I'm made...
              </Text>

          </Stack>
        </Box>
      </Center>
    );
  }