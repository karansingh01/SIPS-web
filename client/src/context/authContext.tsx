import jwtDecode from "jwt-decode";
import React, { useReducer, createContext } from "react";

const initialState = {
    user: null
}

if (localStorage.getItem("token")) {
    const decodedToken = jwtDecode(localStorage.getItem("token")!);

    // checks if token is expired, compares current time to token expiry time
    //@ts-ignore as it's not a string
    if (decodedToken.exp * 1000 < Date.now()) {
        localStorage.removeItem("token");
    } else {
        //@ts-ignore as I know it won't be null
        initialState.user = decodedToken;
    }
}


// create context for auth
const AuthContext = createContext({
    user: null,
    login: (userData: any) => {},
    logout: () => {}
});

// in the case of logging in, we want to return the current state, and change the state of the user to the payload of the action
// in the case of logging out, we want to return the current state, and change the state of the user to null
function authReducer(state: any, action: any) {
    switch (action.type) {
        case "LOGIN":
            return {
                ...state,
                user: action.payload
            }
        case "LOGOUT":
            return {
                ...state,
                user: null
            }
        default:
            return state;
    }
}

// react hook that has a state and a dispatch that comes from useReducer
// dispatch lets us dispatch actions to the reducer
function AuthProvider(props: any) {
    const [state, dispatch] = useReducer(authReducer, initialState);

    // when this login function is called it means we have what we need from the server to log in
    const login = (userData: any) => {
        localStorage.setItem("token", userData.token);
        // dispatch the action to the reducer, user is then set to the payload which is userData
        dispatch({
            type: "LOGIN",
            payload: userData
        });
    }

    // logs out by dispatching the logout action and removing token
    function logout() {
        localStorage.removeItem("token");
        dispatch({ type: "LOGOUT" });
    }

    // returns the provider, which is the context, and the value is the state and the login and logout functions
    // we pass in the props because they are coming from the props that are intially passed into the AuthProvider
    return (
        <AuthContext.Provider
            value={{ user: state.user, login, logout }}
            {...props}
        />
    )
}

export { AuthContext, AuthProvider };