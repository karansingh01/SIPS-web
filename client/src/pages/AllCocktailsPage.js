"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const react_2 = require("@chakra-ui/react");
const Header_1 = __importDefault(require("../components/Header"));
const SearchBar_1 = __importDefault(require("../components/SearchBar"));
const react_3 = require("react");
const DummyData_1 = __importDefault(require("../DummyData"));
const CocktailCardsDisplay_1 = __importDefault(require("../components/CocktailCardsDisplay"));
/**
 * search filter method
 * @param cocktails all cocktails
 * @param query string entered in search field
 * @returns all cocktails containing the search param
 */
const filterCocktails = (cocktails, query) => {
    if (query === "") {
        return cocktails;
    }
    else {
        const filtered = cocktails.filter((cocktail) => {
            return Object.values(cocktail.name)
                .join("")
                .toLowerCase()
                .includes(query.toLowerCase());
        });
        return filtered;
    }
};
const AllCocktailsPage = () => {
    const [query, setQuery] = (0, react_3.useState)("");
    const cocktails = DummyData_1.default;
    const filteredCocktails = filterCocktails(cocktails, query);
    return (<react_2.Flex flexDirection="column">
      <Header_1.default />
      <react_2.Grid transform="translate(0px, -150px)" margin={"30px"} rounded={20} templateRows="repeat(1fr)" templateColumns="repeat(1fr)" gap={4} mt={5}>
        <react_2.GridItem colSpan={4}>
          <SearchBar_1.default q={query} setQuery={setQuery}/>
        </react_2.GridItem>
        <react_2.GridItem colSpan={4}>
          <CocktailCardsDisplay_1.default cocktails={filteredCocktails}/>
        </react_2.GridItem>
      </react_2.Grid>
    </react_2.Flex>);
};
exports.default = AllCocktailsPage;
