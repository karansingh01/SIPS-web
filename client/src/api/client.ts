import {ApolloClient, InMemoryCache} from '@apollo/client';
import { RestLink } from 'apollo-link-rest';

// Set rest-link endpoint
const restLink = new RestLink(
    { uri: 'https://www.thecocktaildb.com/api/json/v1/1/' }
);

// Create apollo client
export const client = new ApolloClient({
    cache: new InMemoryCache(),
    link: restLink,
});


