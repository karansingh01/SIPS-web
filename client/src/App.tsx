import "./App.css";
import { client } from "./api/client";
import { alcoholFilterParam } from "./api/graphql/alcoholFilter";
import { RandomDrinkQuery } from "./api/graphql/randomDrink";
import { VodkaDrinksQuery } from "./api/graphql/vodkaDrinks";
import { GinDrinksQuery } from "./api/graphql/ginDrinks";
import { RumDrinksQuery } from "./api/graphql/rumDrinks";
import { TequilaDrinksQuery } from "./api/graphql/tequilaDrinks";
import reportWebVitals from "./reportWebVitals";
import { HashRouter, Route, Routes } from "react-router-dom";
import { ChakraProvider } from "@chakra-ui/react";
import AllCocktailsPage from "./pages/AllCocktailsPage";
import CocktailRecipePage from "./pages/CocktailRecipePage";
import FavoriteCocktailsPage from "./pages/FavoriteCocktailsPage";
import { gql, useQuery } from "@apollo/client";
import RegisterPage from "./pages/RegisterPage";
import LoginPage from "./pages/LoginPage";
import { ApolloProvider } from "@apollo/client";
import { AuthProvider } from "./context/authContext";
import client1 from "./apolloClient";

// format: response.data.<queryName>.drinks[0].<whateverYouWantToCollect>
// drinks[0] gives you the first drink in the array, which is the only drink since we
// are only querying for one random drink.
client.query({ query: RandomDrinkQuery }).then((response) => {
  console.log("random drink: ", response.data.randomDrink.drinks[0].strDrink);
});

/**
 * Fetch drinks by alcohol (in different files):
 */

client.query({ query: VodkaDrinksQuery }).then((response) => {
  console.log(
    "first vodka drink: ",
    response.data.vodkaDrinks.drinks[0].strDrink
  );
});
client.query({ query: GinDrinksQuery }).then((response) => {
  console.log("first gin drink: ", response.data.ginDrinks.drinks[0].strDrink);
});
client.query({ query: RumDrinksQuery }).then((response) => {
  console.log("first rum drink: ", response.data.rumDrinks.drinks[0].strDrink);
});
client.query({ query: TequilaDrinksQuery }).then((response) => {
  console.log(
    "first tequila drink: ",
    response.data.tequilaDrinks.drinks[0].strDrink
  );
});

console.log("reportwebVitals: ", reportWebVitals);

export default function App() {
  return (
    <AuthProvider>
      <ChakraProvider>
        <ApolloProvider client={client1}>
          <HashRouter>
            <Routes>
              <Route path="/" element={<AllCocktailsPage />} />
              <Route path="/cr" element={<CocktailRecipePage />} />
              <Route path="/fav" element={<FavoriteCocktailsPage />} />
              <Route path="/register" element={<RegisterPage />} />
              <Route path="/login" element={<LoginPage />} />
            </Routes>
          </HashRouter>
        </ApolloProvider>
      </ChakraProvider>
    </AuthProvider>
    /*       <DisplayLocations />
     */
  );
}
