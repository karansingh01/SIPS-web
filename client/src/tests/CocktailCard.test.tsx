import React from "react";
import { render } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import { MockedProvider } from "@apollo/client/testing";
import { HashRouter, Route, Routes } from "react-router-dom";
import { AuthProvider } from "../context/authContext";
import client1 from "../apolloClient";
import AllCocktailsPage from "../pages/AllCocktailsPage";
import App from "../App";
import { ApolloProvider } from "@apollo/client";
import CocktailCard from "../components/CocktailCard";
import { testDrink } from "./test-utils";

beforeEach(() => {
  render(
    <AuthProvider>
      <ChakraProvider>
        <ApolloProvider client={client1}>
          <App />
        </ApolloProvider>
      </ChakraProvider>
    </AuthProvider>
  );
});

test("Snapshot test of CocktailCard", async () => {
  const el = render(<CocktailCard cocktail={testDrink} />);
  /* const el1 = el.findAllByText(""); */
  expect(await el).toMatchSnapshot();
});
