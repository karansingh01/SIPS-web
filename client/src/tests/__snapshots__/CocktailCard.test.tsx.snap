// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of CocktailCard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <p>
        Loading...
      </p>
      <span
        hidden=""
        id="__chakra_env"
      />
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-xqyimh"
      >
        <button
          class="css-1paohjs"
          role="group"
        >
          <div
            class="chakra-stack css-1q0iyqj"
          >
            <p
              class="chakra-text css-1m8qwtx"
            >
              Drink
            </p>
          </div>
          <div
            class="css-1a5ho4p"
          >
            <img
              class="chakra-image css-19s64ds"
              src="test"
            />
          </div>
          <div
            class="chakra-stack css-1282mh9"
          >
            <p
              class="chakra-text css-1cjrhs7"
            >
              Click on me...
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-xqyimh"
    >
      <button
        class="css-1paohjs"
        role="group"
      >
        <div
          class="chakra-stack css-1q0iyqj"
        >
          <p
            class="chakra-text css-1m8qwtx"
          >
            Drink
          </p>
        </div>
        <div
          class="css-1a5ho4p"
        >
          <img
            class="chakra-image css-19s64ds"
            src="test"
          />
        </div>
        <div
          class="chakra-stack css-1282mh9"
        >
          <p
            class="chakra-text css-1cjrhs7"
          >
            Click on me...
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
